libpod/pod.go

```diff
@@ -71,19 +72,167 @@ func (p *Pod) Init() error {
 	return ErrNotImplemented
 }
 
-// Start starts all containers within a pod that are not already running
+// Start starts all containers within a pod
+// Containers that are already running or have been paused are ignored
+// If an error is encountered starting any container, Start() will cease
+// starting containers and immediately report an error
+// Start() is not an atomic operation - if an error is reported, containers that
+// have already started will remain running
 func (p *Pod) Start() error {
-	return ErrNotImplemented
+	p.lock.Lock()
+	defer p.lock.Unlock()
+
+	if !p.valid {
+		return ErrPodRemoved
+	}
+
+	allCtrs, err := p.runtime.state.PodContainers(p)
+	if err != nil {
+		return err
+	}
+
+	// We need to lock all the containers
```

Comment:
I'm concerned about trying to lock all the containers and then perform operations on a subset.  Can't we lock just the ones we're going to do an operation on?  Same thought in the other functions.

Reply:
We need to lock them to update their status, which we need to know whether we're going to operate on them. We could potentially do a quick unlock here if we're not going to end up mutating them, but the logic for that could get messy?